siliconcompiler/core.py

```diff
@@ -4008,6 +4008,8 @@ def run(self):
                 local_dir = self.get('option','builddir')
                 self.read_manifest(cfg, clobber=True, clear=True)
                 self.set('option', 'builddir', local_dir)
+                # Un-set steplist so 'show'/etc flows will work on returned results.
+                self.unset('option', 'steplist')
```

Comment:
Is this necessarily something we want to do? I think if the user sets `['option', 'steplist']`, the setting should persist after `run()` is called. Perhaps a better thing would be to store whether steplist has been set/its value before calling `remote_preprocess()` and then restore the state of the parameter at the end (unset() if it wasn't set, otherwise reset it to the value).

Reply:
This is all under an `if self.get('option','remote'):` block, so it won't get run for local builds with user-defined steplists.

I suppose it's an open question as to whether we should let people define a subset of steps to run in the remote flow. It might be useful if people want to experiment with early steps like floorplan/placement.

Reply:
Got it - I forgot that steplists aren't supported with the remote run at the moment. This seems fine then!